src/main/java/training/endava/app/repository/PersonRepository.java

```diff
@@ -1,8 +1,94 @@
 package training.endava.app.repository;
 
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import training.endava.app.domain.Person;
+import training.endava.app.exceptions.BadRequestException;
+import training.endava.app.exceptions.BaseException;
+import training.endava.app.exceptions.PersonAlreadyExists;
+import training.endava.app.exceptions.PersonNotFoundException;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
 
-
+@Repository
 public class PersonRepository {
 
+    private List<Person> personList = null;
+
+    public PersonRepository(){
+        this.personList = new ArrayList<>();
+        Person p1 = new Person(1, "Irimia Cosmin", "0756735362", "Iasi");
+        Person p2 = new Person(2, "Matei Madalin", "07689394394", "Bacau");
+        Person p3 = new Person(3, "Bucevschi Alexandru", "07539192833", "Botosani");
+        Person p4 = new Person(4, "Trinca Ioana", "0749287483", "Iasi");
+        Person p5 = new Person(5, "Olariu Madalina", "07516738483", "Vaslui");
+        Person p6 = new Person(6, "Cineva Marcel", "07576738483", "Bacau");
+        this.personList.addAll(Arrays.asList(p1,p2,p3,p4,p5,p6));
+    }
+
+    public Person getPersonById(Integer id){
+        Optional<Person> p = this.personList.stream().filter(person -> person.getId().equals(id)).findFirst();
+        if(p.isPresent()){
```

Comment:
Try to change the p.isPresent() by returning orElse give null back, or more nicely, throw the NotFoundPerson exception